src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,120 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a task list.
+ * Contains all current tasks and contain methods to add/delete/mark/unmark a specific task.
+ */
+public class TaskList {
+    protected static ArrayList<Task> toDo;
+
+    /**
+     * Constructs a new <code>TaskList</code> object.
+     */
+    public TaskList() {
+        toDo = new ArrayList<>();
+    }
+
+    /**
+     * Adds a new task to the task list from user input.
+     * @param task the task to add into task list.
+     * @param type type of task added.
+     */
+    public static void add(Task task, String type) {
+        toDo.add(task);
+        String description = task.getDescription();
+        Ui.successfulAdd(type, description, toDo.size());
+        Storage.save();
+    }
+
+    /**
+     * Adds a new task to the task list from the hard disk when chatbot starts up.
+     * @param task the task from the file in the hard disk to be loaded into the task list.
+     */
+    public static void add(Task task) {
+        toDo.add(task);
+    }
+
+    /**
+     * Deletes a specific task from the task list.
+     * @param target the index of the task in the task list that should be deleted.
+     * @throws DukeException throws exception when the index of the task given is larger than the size of the task list.
```

Comment:
Just on the brink of the hard limit for line character count (120). Would be best to separate into two lines to improve readability.

Reply:
Will do, thanks for the suggestion